### PR TITLE
[FLINK-19883][table] Support "IF EXISTS" in DDL for ALTER TABLE

### PR DESCRIPTION
**Pull request by LadyForest**
## What is the purpose of the change

This pull request is a subtask of FLINK-21634 and aims to support the `ALTER TALBE IF EXISTS <table_identifier> ...` statement.

## Brief change log

- 6167b501 extends the parser template to support `IF EXISTS` and adjusts `SqlAlterTable` to add `ifExists` to the constructor params. During the operation conversion phase, `SqlToOperationConverter` will convert the alter statement to a `NopOperation` to be ignored if the table does not exist and `IF EXISTS` is specified.

## Verifying this change

This change added tests and can be verified as follows:

- FlinkSqlParserImplTest
- SqlToOperationConverterTest
- table.q

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): No
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: No
  - The serializers: No
  - The runtime per-record code paths (performance sensitive): No
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: No
  - The S3 file system connector: No

## Documentation

  - Does this pull request introduces a new feature? Yes
  - If yes, how is the feature documented? It'll be documented after FLINK-22320 finishes

### CODE REVIEW

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlAlterTable.java [hunk 1]
```diff
@@ -98,6 +111,15 @@ public LinkedHashMap<String, String> getPartitionKVs() {
         return SqlPartitionUtils.getPartitionKVs(getPartitionSpec());
     }
 
+    /**
+     * Whether to ignore the error if the table doesn't exist.
+     *
+     * @return true when IF EXISTS is specified.
+     */
+    public boolean ignoreIfNotExists() {
```
Comment (fsk119) [inline flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlAlterTable.java:119]: BTW, I think we should keep the code style same. In the `SqlAlterFunction`, we use `isIfExists` as method name.

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I left some minor comments.

**Review by fsk119 — COMMENTED** [review]
Thanks for your update. I left some minor suggestions.

**Comment by LadyForest** [comment]
@flinkbot run azure

**Review by fsk119 — APPROVED** [review]
LGTM

**Comment by fsk119** [comment]
@flinkbot run azure

**Pull request merged**